x-pack/legacy/plugins/advanced_ui_actions/public/np_ready/public/custom_time_range_action.test.ts

```diff
@@ -0,0 +1,260 @@
+
+  const container = new TimeRangeContainer(
+    {
+      timeRange: { from: 'now-15m', to: 'now' },
+      panels: {
+        '1': {
+          type: TIME_RANGE_EMBEDDABLE,
+          explicitInput: {
+            id: '1',
+          },
+        },
+        '2': {
+          type: TIME_RANGE_EMBEDDABLE,
+          explicitInput: {
+            id: '2',
+          },
+        },
+      },
+      id: '123',
+    },
+    (() => {}) as any
+  );
+
+  await container.untilEmbeddableLoaded('1');
+  await container.untilEmbeddableLoaded('2');
+
+  const child1 = container.getChild<TimeRangeEmbeddable>('1');
+  expect(child1).toBeDefined();
+  expect(child1.getInput().timeRange).toEqual({ from: 'now-15m', to: 'now' });
+
+  const child2 = container.getChild<TimeRangeEmbeddable>('2');
+  expect(child2).toBeDefined();
+  expect(child2.getInput().timeRange).toEqual({ from: 'now-15m', to: 'now' });
+
+  const start = coreMock.createStart();
+  const overlayMock = start.overlays;
+  (overlayMock.openModal as any).mockClear();
+  new CustomTimeRangeAction({
+    openModal: start.overlays.openModal,
+    commonlyUsedRanges: [],
+    dateFormat: 'MM YYY',
+  }).execute({
+    embeddable: child1,
+  });
+
+  const openModal = (overlayMock.openModal as any).mock.calls[0][0];
+
+  const wrapper = mount(openModal);
+  wrapper.setState({ timeRange: { from: 'now-30days', to: 'now-29days' } });
+
+  findTestSubject(wrapper, 'addPerPanelTimeRangeButton').simulate('click');
+
+  const subscription = Rx.merge(container.getOutput$(), container.getInput$())
+    .pipe(skip(2))
+    .subscribe(() => {
+      expect(child1.getInput().timeRange).toEqual({ from: 'now-30days', to: 'now-29days' });
+      expect(child2.getInput().timeRange).toEqual({ from: 'now-30m', to: 'now-1m' });
+      subscription.unsubscribe();
+      done();
+    });
+
+  container.updateInput({ timeRange: { from: 'now-30m', to: 'now-1m' } });
+});
+
+test('Removing custom time range action resets embeddable back to container time', async done => {
+  const embeddableFactories = new Map<string, EmbeddableFactory>();
+  embeddableFactories.set(TIME_RANGE_EMBEDDABLE, new TimeRangeEmbeddableFactory());
+
+  const container = new TimeRangeContainer(
+    {
+      timeRange: { from: 'now-15m', to: 'now' },
+      panels: {
+        '1': {
+          type: TIME_RANGE_EMBEDDABLE,
+          explicitInput: {
+            id: '1',
+          },
+        },
+        '2': {
+          type: TIME_RANGE_EMBEDDABLE,
+          explicitInput: {
+            id: '2',
+          },
+        },
+      },
+      id: '123',
+    },
+    (() => {}) as any
+  );
+
+  await container.untilEmbeddableLoaded('1');
+  await container.untilEmbeddableLoaded('2');
+
+  const child1 = container.getChild<TimeRangeEmbeddable>('1');
+  const child2 = container.getChild<TimeRangeEmbeddable>('2');
+
+  const start = coreMock.createStart();
+  const overlayMock = start.overlays;
+  (overlayMock.openModal as any).mockClear();
```

Comment:
I believe you don't need this `any`. As mocks by Core are such that they have both types: (1) type of the modal; (2) type of the `jest.fn()`.